pulsar-functions/localrun/src/main/java/org/apache/pulsar/functions/LocalRunner.java

```diff
@@ -544,6 +552,11 @@ private void startThreadedMode(org.apache.pulsar.functions.proto.Function.Functi
         if (functionConfig != null && functionConfig.getExposePulsarAdminClientEnabled() != null) {
             exposePulsarAdminClientEnabled = functionConfig.getExposePulsarAdminClientEnabled();
         }
+
+        // Collector Registry for prometheus metrics
+        CollectorRegistry collectorRegistry = new CollectorRegistry();
```

Comment:
If there is just one `CollectorRegistry` shared among threads (assuming parallelism > 1 here), won't each metrics port serve the same metrics?

Reply:
@michaeljmarshall yup thus after some thinking, I can up with a new PR:

https://github.com/apache/pulsar/pull/10208

For running instances as threads with in the local run JVM, it doesn't make sense to run a metrics server per instance.  Better user experience to just run one metrics server and expose all of the metrics for all instances on the same port.

Reply:
>  Better user experience to just run one metrics server and expose all of the metrics for all instances on the same port.

I agree. Although, it looks like the new PR still has metrics at `p` ports where `p = parallelism`. I'll add a note to the new PR too, but wanted to comment here, where you mentioned a single port.